common/setups/hybrid/gmm_system.py

```diff
@@ -0,0 +1,878 @@
+import i6_core.lda as lda
+import i6_core.meta as meta
+import i6_core.sat as sat
+import i6_core.rasr as rasr
+import i6_core.util as util
+import i6_core.vtln as vtln
+
+from .util import (
+    GmmDataInput,
+    GmmInitArgs,
+    GmmMonophoneArgs,
+    GmmTriphoneArgs,
+    GmmVtlnArgs,
+    GmmSatArgs,
+    GmmVtlnSatArgs,
+)
+
+# -------------------- Init --------------------
+
+Path = tk.setup_path(__package__)
+
+
+# -------------------- System --------------------
+
+
+class GmmSystem(meta.System):
+    """
+    - 3 corpora types: train, dev and test
+    - only train corpora will be aligned
+    - dev corpora for tuning
+    - test corpora for final eval
+
+    to create beforehand:
+    - corpora: name and i6_core.meta.system.Corpus
+    - lexicon
+    - lm
+
+    settings needed:
+    - am
+    - lm
+    - lexicon
+    - feature extraction
+    - linear alignement
+    - monophone training
+    - monophone recognition
+    - triphone training
+    - triphone recognition
+    - vtln training
+    - vtln recognition
+    - sat training
+    - sat recognition
+    - vtln+sat training
+    - vtln+sat recognition
+    """
+
+    def __init__(self):
+        super().__init__()
+
+        self.crp["base"].python_home = gs.RASR_PYTHON_HOME
+        self.crp["base"].python_program_name = gs.RASR_PYTHON_EXE
+
+        self.gmm_init_args = None
+        self.gmm_monophone_args = None
+        self.gmm_triphone_args = None
+        self.gmm_vtln_args = None
+        self.gmm_sat_args = None
+        self.gmm_vtln_sat_args = None
+
+        self.train_corpora = []
+        self.dev_corpora = []
+        self.test_corpora = []
+
+        self.corpora = {}
+        self.concurrent = {}
+
+        self.cart_questions = None
+        self.lda_matrices = {}
+        self.cart_trees = {}
+
+        self.vtln_files = {}
+
+        self.default_align_keep_values = {
+            "default": 5,
+            "selected": gs.JOB_DEFAULT_KEEP_VALUE,
+        }
+
+    # -------------------- Setup --------------------
+    def init_system(
+        self,
+        gmm_init_args: GmmInitArgs,
+        gmm_monophone_args: GmmMonophoneArgs,
+        gmm_triphone_args: GmmTriphoneArgs,
+        gmm_vtln_args: GmmVtlnArgs,
+        gmm_sat_args: GmmSatArgs,
+        gmm_vtln_sat_args: GmmVtlnSatArgs,
+        train_data: Dict[str, GmmDataInput],
+        dev_data: Dict[str, GmmDataInput],
+        test_data: Dict[str, GmmDataInput],
+    ):
+        """
+        :param gmm_init_args: parameters for am and feature extraction
+        :param gmm_monophone_args: parameters for lin align and monophone
+        :param gmm_triphone_args: parameters for cart and triphone
+        :param gmm_vtln_args: parameters for VTLN
+        :param gmm_sat_args: parameters for SAT
+        :param gmm_vtln_sat_args: parameters for VTLN+SAT
+        :param dict[str, GmmInput] train_data:
+        :param dict[str, GmmInput] dev_data:
+        :param dict[str, GmmInput] test_data:
```

Comment:
This is inconsistent to the type documentation above (`GmmDataInput` vs `GmmInput`), so sth is wrong.

But also, I don't think we need to have such information redundantly. I would suggest to only use type annotations.

Also, I don't really like documentation which doesn't say anything, like `gmm_vtln_args: parameters for VTLN`. Just leave it away, which reduces the time to read it, and thus makes it more clear in the end.

Also, if the type is already documented via type annotations, I think you don't need to mention the parameter at all in the docstring anymore (unless you want to really add further information/documentation for it).
(In RETURNN, we want to have all parameters listed, because this is always the way we document types, because RETURNN doesn't use type annotations yet, because Python 2 compatibility.)